src/Data/BigInt.purs

```diff
@@ -37,23 +37,37 @@ import Partial.Unsafe (unsafePartial)
 -- | An arbitrary length integer.
 foreign import data BigInt :: Type
 
-type BaseDigits = {
-    value :: NonEmptyArray Int
+type BaseDigits =
+  { value :: NonEmptyArray Int
   , isNegative :: Boolean
-}
+  }
 
 -- | FFI wrapper to parse a String in a given base representation into a BigInt.
-foreign import fromBase' :: forall a. (a -> Maybe a)
-                         -> Maybe a
-                         -> Int
-                         -> String
-                         -> Maybe BigInt
+foreign import fromBase'
+  :: forall a
+   . (a -> Maybe a)
+  -> Maybe a
+  -> Int
+  -> String
+  -> Maybe BigInt
+
+-- | Truncate a Number to an Int (towards zero).
+-- | `Infinity`, `-Infinity`, and `NaN` pass through.
+foreign import numberToString :: Number -> String
+
+-- | Convert an Int to a String
+foreign import intToString :: Int -> String
 
 -- | Convert an integer to a BigInt.
-foreign import fromInt :: Int -> BigInt
+fromInt :: Int -> BigInt
+fromInt a = unsafePartial
```

Comment:
I consolidated on `fromBase'` when I was thinking `fromInt` could fail as well, but that wasn't correct. Before I did that though, I skimmed the `big-integer` code to see how it handled strings vs numbers. Since it converts numbers to strings and uses the string implementation for the real logic, it seemed simpler to focus on a smaller set of FFI.

Reply:
https://github.com/peterolson/BigInteger.js/blob/master/BigInteger.js#L1237

Reply:
okay .. but it also includes some additional logic (https://github.com/peterolson/BigInteger.js/blob/5d47b8c2e704218348997f22efbdc5a8a175e2e0/BigInteger.js#L1233-L1236) and things could change in the future. In general, it is great to write as little FFI code as necessary, but purescript-bigints is just a wrapper library that fundamentally assumes that the underlying JavaScript library works correctly.